### DISCUSSION TITLE
Add custom field relation(new entity/gql type in plugin) in orderLine

### DISCUSSION THREAD

**Discussion by lucasrochaa** [vendurehq/vendure#1538] [Q&A]
I created a customField relation with a entity of my ReceiptPlugin

**vendure-config.ts**

`OrderLine: [
{name: 'receipt', type: 'relation', public: true, entity: Receipt, graphQLType: 'Receipt', eager: false,},
],

**receipt-plugin.ts**
`
type Receipt implements Node {
                id: ID!
                numDocumento: String!
                idTransacao: String!
                dataPgtoEfetuado: DateTime!
                valorNominal: Int!
                valorTarifa: Int!
                valorTotal: Int!
                protocolo: String!
                nomePagador: String!
                cpf: String!
            }
`
when i start the server i get:
Error: The GraphQL type "Receipt" specified by the OrderLine.receipt custom field does not exist

and in the database, i've got :

customFields__fix_relational_custom_fields__  (new column in order_line) and have a description:

A work-around needed when only relational custom fields are defined on an entity

What i'am missing?

thanx!

**Comment by michaelbromley** [comment]
Hi,
Is the `Receipt` type defined in the Shop API as well as the Admin API? Because you set the custom field to `public: true`, Vendure is going to expose it in the Shop API, so you need to ensure that the corresponding GraphQL type is also being passed to the `shopApiExtensions` in your plugin metadata.

**Reply by lucasrochaa** [reply]
Thanx for the answer, i was declaring the type in the plugin in shopApiExtensions, and not in adminApiExtensions.
